test/TestPublicResolver.js

```diff
@@ -659,6 +659,40 @@ contract('PublicResolver', function (accounts) {
             assert.equal(await resolver.addr(node), accounts[0]);
         });
     });
+
+    describe('multicall', async () => {
+        it('allows setting multiple fields', async () => {
+          var addrSet = resolver.contract.methods['setAddr(bytes32,address)'](node, accounts[1]).encodeABI();
+          var textSet = resolver.contract.methods.setText(node, "url", "https://ethereum.org/").encodeABI();
+          var tx = await resolver.multicall([addrSet, textSet], {from: accounts[0]});
+
+          assert.equal(tx.logs.length, 3);
+          assert.equal(tx.logs[0].event, "AddressChanged");
+          assert.equal(tx.logs[0].args.node, node);
+          assert.equal(tx.logs[0].args.newAddress, accounts[1].toLowerCase());
+          assert.equal(tx.logs[1].event, "AddrChanged");
+          assert.equal(tx.logs[1].args.node, node);
+          assert.equal(tx.logs[1].args.a, accounts[1]);
+          assert.equal(tx.logs[2].event, "TextChanged");
+          assert.equal(tx.logs[2].args.node, node);
+          assert.equal(tx.logs[2].args.key, "url");
+
+          assert.equal(await resolver.methods['addr(bytes32)'](node), accounts[1]);
+          assert.equal(await resolver.text(node, "url"), "https://ethereum.org/");
+        });
+
+        // Disabled because web3 doesn't have a function for decoding the return ABI
```

Comment:
You could try decoding with ethersjs which I think it does support.

```Meta-classes create JavaScript objects from any contract ABI, including ABIv2 and Human-Readable ABI```

https://docs.ethers.io/ethers.js/html/index.html?highlight=abiv2
https://blog.ricmoo.com/solidity-abiv2-a-foray-into-the-experimental-a6afd3d47185

Reply:
I don't really want to include a new library just for this one test. I'll work around it.